infra/src/test/kotlin/fi/fta/geoviite/infra/publication/PublicationServiceIT.kt

```diff
@@ -2979,6 +3002,241 @@ class PublicationServiceIT @Autowired constructor(
+        val testOfficialContext = testDBService.testContext(testBranch, PublicationState.OFFICIAL)
+
+        testDraftContext.insert(
+            asDesignDraft(
+                mainOfficialContext
+                    .fetch(trackNumber)!!
+                    .copy(number = TrackNumber("edited")), testBranch.designId
+            )
+        )
+        testDraftContext.insert(
+            asDesignDraft(
+                mainOfficialContext
+                    .fetch(referenceLine)!!
+                    .copy(startAddress = TrackMeter("0001+0123")), testBranch.designId
+            )
+        )
+        testDraftContext.insert(
+            asDesignDraft(
+                mainOfficialContext
+                    .fetch(locationTrack)!!
+                    .copy(name = AlignmentName("edited")), testBranch.designId
+            )
+        )
+        testDraftContext.insert(
+            asDesignDraft(
+                mainOfficialContext.fetch(kmPost)!!.copy(kmNumber = KmNumber(123)),
+                testBranch.designId
+            )
+        )
+        testDraftContext.insert(
+            asDesignDraft(
+                mainOfficialContext.fetch(switch)!!.copy(name = SwitchName("edited")),
+                testBranch.designId
+            )
+        )
+
+        publishAndVerify(
+            testBranch, publicationRequest(
+                trackNumbers = listOf(trackNumber),
+                referenceLines = listOf(referenceLine),
+                kmPosts = listOf(kmPost),
+                locationTracks = listOf(locationTrack),
+                switches = listOf(switch),
+            )
+        )
+        mainDraftContext.insert(asMainDraft(testOfficialContext.fetch(trackNumber)!!))
+        mainDraftContext.insert(asMainDraft(testOfficialContext.fetch(referenceLine)!!))
+        mainDraftContext.insert(asMainDraft(testOfficialContext.fetch(locationTrack)!!))
+        mainDraftContext.insert(asMainDraft(testOfficialContext.fetch(kmPost)!!))
+        mainDraftContext.insert(asMainDraft(testOfficialContext.fetch(switch)!!))
+
+        publishAndVerify(
+            LayoutBranch.main, publicationRequest(
+                trackNumbers = listOf(trackNumber),
+                referenceLines = listOf(referenceLine),
+                kmPosts = listOf(kmPost),
+                locationTracks = listOf(locationTrack),
+                switches = listOf(switch),
+            )
+        )
+
+        assertEquals("edited", mainOfficialContext.fetch(trackNumber)!!.number.toString())
+        assertEquals("0001+0123", mainOfficialContext.fetch(referenceLine)!!.startAddress.toString())
+        assertEquals("edited", mainOfficialContext.fetch(locationTrack)!!.name.toString())
+        assertEquals("0123", mainOfficialContext.fetch(kmPost)!!.kmNumber.toString())
+        assertEquals("edited", mainOfficialContext.fetch(switch)!!.name.toString())
+    }
+
+    @Test
+    fun `create in main and alter in design twice before updating main`() {
```

Comment:
Jos `alter twice in design` onnistuu niin luulisi että myös `alter once in design` onnistuu -> näistä kahdesta voisi jättää jommankumman ja poistaa toisen